README.md

```diff
@@ -1,104 +1,130 @@
+ Full Documentation can be found [here](https://docs.threshold.network/app-development/threshold-access-control-tac).
 
-## Disclaimer
+> :warning: 
+> `nucypher-ts` is under [active development](https://github.com/nucypher/nucypher-ts/pulls).
+>
+> SDK does not support policy revocation.
+>
+> We expect breaking changes.
+# Get Started with TAC
 
-**This is a work in progress**
+The following interactive tutorial is a quick way for developers to familiarize themselves with the Threshold Access Control service. Before jumping into the following, we recommend reading this introduction to [Conditions Based decryption](https://docs.threshold.network/fundamentals/threshold-access-control/conditions-based-decryption-cbd) and its key concepts. CBD is the primary technology underpinning Threshold Access Control and directly enables data sharing predicated on the fulfillment of predefined conditions.
 
-- SDK does not support policy revocation.
-- We expect breaking changes.
+## 1. Install nucypher-ts
 
-## Supported networks
+We'll kick things off by installing `nucypher-ts` – a TypeScript library for interacting with access control functionality in the browser. The APIs for leveraging most TAC functionality are contained in nucypher-ts.
 
-`nucypher-ts` is available on the following networks:
+```
+yarn add @nucypher/nucypher-ts
+```
+
+## 2. Build a Cohort
 
-- Polygon
-- Mumbai (Polygon testnet)
+Next, we'll parametrize a _Cohort_ to correspond to our risk preferences. Cohort objects delineate the group of independent nodes that will collectively provide access control service to a given data sharing flow. _Threshold_ and _Shares_ are two parameters used to construct a Cohort. For example, a `3-of-5` Cohort requires responses – the delivery of shares – from a minimum of 3 out of a total of 5 Cohort members in order to reconstruct the original plaintext data.
 
-## Usage
+We generate a Cohort object by:
 
-Run with:
+```javascript
+import { Cohort } from '@nucypher/nucypher-ts';
 
-```bash
-yarn install
-yarn test
-yarn build
+const config = {
+  threshold: 3,
+  shares: 5,
+  porterUri: 'https://porter-lynx.nucypher.community',
+};
+const newCohort = await Cohort.create(config);
 ```
 
-## Development
+Notice that we have also provided a `porterUri`. **Porter is a web-based service that interacts with nodes on the network on behalf of applications – an "Infura for TAC".** In this example, we've chosen a `lynx` (testnet) Porter endpoint.
 
-Install git hooks
+## 3. Create Conditions
 
-```bash
-npx husky install
-```
+We will now specify the conditions on which data access will be predicated – i.e. what will the data requester need to prove in order to gain decryption rights. In this tutorial, nodes will check that the requester owns a specific ERC721 NFT:
 
-Generate contract typings
+```javascript
+import { Conditions } from '@nucypher/nucypher-ts';
 
-```bash
-yarn typechain
+const NFTOwnership = new Conditions.ERC721Ownership({
```

Comment:
Noticed this in Manu's gitbook change as well. Is there a reason that ERC721Ownership takes a dict instead of just parameters?

Reply:
^ (cc @piotr-roslaniec as well)

Reply:
> prevents misuse (like messing up the ordering of parameters).

Not sure I grasp the misuse/ordering concern/resolution - interested to understand this a bit more. Could you elaborate?

From a high-level, it seems cumbersome that a dict needs to be provided to a constructor for something that's a convenience wrapper (all configuration values don't explicitly need to be provided)...as opposed to using individual parameters. The individual parameters are simply entries from the dict.

Basically the difference would be doing something like this instead:
```
new Conditions.ERC721Ownership(
    contractAddress=...,
    chain=...,
    parameters=[...],
)
```

> we initially assumed that all conditions are JSONs and we validate their structure based on that.

The *full* definition of a condition is JSON - and presumably that's what the `fromJSON(...)` method, and private constructors on the base condition classes were for.

However, for the constructor of a convenience wrapper that takes incomplete inputs (because the rest is added in the backgrounds), it can presumably generate the JSON anyway in its underlying constructor, and then validate everything appropriately.

Reply:
It's not possible to pass named parameters in JS/TS.
So a JS/TS equivalent of this code
```
new Conditions.ERC721Ownership(
    contractAddress=...,
    chain=...,
    parameters=[...],
)
```
would be
```
new Conditions.ERC721Ownership({
    contractAddress: ...,
    chain: ...,
    parameters: [...],
})
```

> (...) it can presumably generate the JSON anyway in its underlying constructor, and then validate everything appropriately?

Yes, it's possible. I'm not hellbent on this design. We don't expose schemas, only validation methods. So the only way to generate a correct condition is to try to make one. Also, notice how JS users don't take advantage of typing.

Reply:
Thanks for the background info. Clearly I need to brush up on my ts/js 😅 .